controllers/controller_test.go

```diff
@@ -230,24 +233,26 @@ var _ = Describe("snr Controller", func() {
 
 				verifyNodeBackup()
 
-				verifyFinalizerExists()
-
-				verifyNoExecuteTaintExist()
-
 				verifyNoWatchdogFood()
 
 				verifySelfNodeRemediationPodDoesntExist()
 
 				verifyVaDeleted(vaName)
 
+				verifyFinalizerExists()
```

Comment:
why did you move these 2 verifies down here? Finalizer and taint are set early in the remdiaton process, so IMHO we should also test it early

Reply:
Due to the code change, now they are deleted at a later stage (only when NHC triggers deletion) so I thought it make sense for the validation code to reflect that.

Reply:
I can't follow. This test ensures it exists, not that it's deleted... 🤔

Reply:
You are correct, however as a rule of we should verify something exit at the latest possible point.

For example let's say that we have a bug in the code and the taint is deleted just after Watchdog is stopped (`verifyNoWatchdogFood`) in the previous order we could have easily missed that bug but in current order it will be caught.

Reply:
>  we should verify something exist at the latest possible point

Sorry but I disagree. Setting the finalizer and taint is the first thig we need to do, so IMHO it should also be the first thing to test. 

> we have a bug in the code and the taint is deleted just after Watchdog is stopped

Then this is a 2nd test. Call verify again instead of moving it around 🙂

Reply:
>Setting the finalizer and taint is the first thig we need to do, so IMHO it should also be the first thing to test.

IMO it makes better sense to verify according to removal order (i.e taint is removed last so verified it still exist last).
We can catch more bugs using that approach (per previous example), so the value in that approach is evident beyond personal taste on what is more readable or not (which is far more arguable).

>Then this is a 2nd test. Call verify again instead of moving it around :slightly_smiling_face: 

Since our tests are meant to deal with async code (i.e using `Eventually` to verify) I don't see any concrete value for that (i.e bugs that we will catch from having double taint verification vs just one at the end) please let me know if I'm missing something here.